app/services/sessions/retrieve.rb

```diff
@@ -0,0 +1,45 @@
+module Sessions
+  class Retrieve
+    attr_reader :token, :email, :password
+
+    def initialize(token: nil, email: nil, password: nil)
+      @token, @email, @password = token, email, password
+    end
+
+    def process
+      if session.present?
+        get_session
+      else
+        create_session
+      end
```

Comment:
I think this class looks very good 🤓 , however I believe that the fact that we have the `get_session` and `create_session` tells me that this class has more than 1 responsibility. What do you think?